### ISSUE TITLE
LDAP - avatar in format other than jpg/png throws exception

### ISSUE THREAD

**Issue by 11mariom** [owncloud/core#17232]
Hi,
when using LDAP with OwnCloud lib/private/avatar.php ( https://github.com/owncloud/core/blob/master/lib/private/avatar.php#L104 ) could make instance unusable for some users. If they have avatar in other format than jpg or png - they have 500 internal server error, in logs you just see exception from avatar.php.

Exception:
```javascript
{"Message":"Unknown filetype","Code":0,"Trace":"#0 \/var\/www\/html\/owncloud\/apps\/user_ldap\/lib\/user\/user.php(346): OC_Avatar->set(Object(OCP\\Image))\n#1 \/var\/www\/html\/owncloud\/apps\/user_ldap\/lib\/user\/user.php(319): OCA\\user_ldap\\lib\\user\\User->setOwnCloudAvatar()\n#2 \/var\/www\/html\/owncloud\/apps\/user_ldap\/lib\/user\/user.php(134): OCA\\user_ldap\\lib\\user\\User->updateAvatar()\n#3 \/var\/www\/html\/owncloud\/apps\/user_ldap\/user_ldap.php(213): OCA\\user_ldap\\lib\\user\\User->update()\n#4 \/var\/www\/html\/owncloud\/lib\/private\/user\/manager.php(113): OCA\\user_ldap\\USER_LDAP->userExists('user')\n#5 \/var\/www\/html\/owncloud\/lib\/private\/user\/manager.php(142): OC\\User\\Manager->get('user')\n#6 \/var\/www\/html\/owncloud\/lib\/private\/user.php(574): OC\\User\\Manager->userExists('user')\n#7 \/var\/www\/html\/owncloud\/lib\/private\/user.php(334): OC_User::userExists('user')\n#8 \/var\/www\/html\/owncloud\/lib\/base.php(774): OC_User::isLoggedIn()\n#9 \/var\/www\/html\/owncloud\/index.php(36): OC::handleRequest()\n#10 {main}","File":"\/var\/www\/html\/owncloud\/lib\/private\/avatar.php","Line":66} 
```

Steps to reproduce:
1. Use LDAP
2. Create user in LDAP with f.e. gif avatar.
3. Try to login to OwnCloud

or
1. Use LDAP
2. Create user in LDAP
3. Login to OwnCloud with that user
4. Set in LDAP gif avatar
5. Every link user opens - will give 500 Internal server error.

**Comment by DeepDiver1975** [comment]
@blizzz THX

**Comment by blizzz** [comment]
Thank you for reporting. 

Interesting thing is that the image  was checked earlier in that function to be valid. But Avatar has more restrictions. We just need to catch the exception there and leave a log message.

**Comment by DeepDiver1975** [comment]
#17464 is still under review to be merged into master - no backport yet available -> 8.0.7
